### ISSUE TITLE
Broken depth targets: checkpoint key mismatch silently skips `depth_patch_embed` weights

### ISSUE THREAD

**Issue by myshrry** [Robbyant/lingbot-vla-v2#33]
Hi, congratulations on the awesome work and the open-source release!

While investigating why the GT depth maps looked much worse than the predicted ones, it seems like a bug in the released `depth/model.pt` checkpoint.

## Summary

The checkpoint `lingbot-vla/lingbot-vla-v2-6b/depth/model.pt` contains two keys named:

- `encoder.backbone.depth_mask_patch_embed.proj.weight`
- `encoder.backbone.depth_mask_patch_embed.proj.bias`

However, in `DinoVisionTransformer` (`lingbotvla/models/vla/vision_models/lingbot-depth/mdm/model/dinov2_rgbd/models/vision_transformer.py`), the corresponding module is created as `depth_patch_embed` (without `_mask`):

```python
if self.depth_emb_mode == 'conv_1c':
    self.depth_patch_embed = embed_layer(img_size=img_size, patch_size=patch_size, in_chans=1, embed_dim=embed_dim)
else:
    self.depth_patch_embed = None
```

Since `MDMModel.from_pretrained` (`lingbotvla/models/vla/vision_models/lingbot-depth/mdm/model/v2.py`) loads the state dict with `strict=False`, the mismatched keys are **silently skipped** and `depth_patch_embed` keeps its random initialization. As a result, the depth branch of the encoder receives randomly-embedded depth input, and every feature produced downstream (e.g. `depth_targets` from `get_depth_target`) is effectively garbage.

## Proof / Reproduction

When decoding `depth_targets` back to dense depth maps, the GT depth is significantly worse than the depth predicted by `robbyant/lingbot-vla-v2-6b-robotwin`. Renaming the keys before `load_state_dict` immediately fixes the GT depth:

```python
state_dict = checkpoint['model']
state_dict = {
    (k.replace('depth_mask_patch_embed', 'depth_patch_embed') if 'depth_mask_patch_embed' in k else k): v
    for k, v in state_dict.items()
}
model.load_state_dict(state_dict, strict=False)
```

With this change, the decoded GT depth maps look correct, and a quick sanity check confirms the loaded weights now match the checkpoint (no missing `depth_patch_embed` keys).

## Impact

- All depth-alignment targets (`depth_targets` / `future_depth_targets`) computed during training and visualization are produced by a randomly-initialized depth embedding.
- The depth loss is therefore optimizing toward meaningless targets (weight is small, e.g. 0.004, so the impact on action learning is limited, but the auxiliary depth alignment is broken).

## Suggested Fix

Either re-export the checkpoint with corrected key names, or apply the rename inside `MDMModel.from_pretrained` before `load_state_dict` for backward compatibility.

**Comment by LuFan31** [comment]
Thanks for the detailed investigation and reproduction!

This key mismatch did exist in an earlier version of the code: the released checkpoint uses `depth_mask_patch_embed`, while the corresponding module in the code was previously named `depth_patch_embed`. Since the checkpoint is loaded with `strict=False`, this could indeed cause these weights to be skipped silently.

We have already fixed this issue in commit [`[311a4d2](https://github.com/Robbyant/lingbot-vla-v2/commit/311a4d2d981a9775741b283d412372ed054a1603)`](https://github.com/Robbyant/lingbot-vla-v2/commit/311a4d2d981a9775741b283d412372ed054a1603) by renaming the module in the code to `depth_mask_patch_embed`, so that it is consistent with the released checkpoint.

Therefore, the released checkpoint itself does not need to be re-exported, and no additional key remapping is required with the current `main` branch.

Please pull the latest version of the repository and try again. If you still encounter the issue with the latest `main`, feel free to let us know.

Thanks again for carefully checking this and reporting it!

**Comment by myshrry** [comment]
Thanks for the detailed explanation! Turns out the root cause was on my side — I hadn't pulled the latest version of the repository. Sorry for the confusion.
